src/net.cpp

```diff
@@ -2634,9 +2634,10 @@ void DumpBanlist()
 
     CBanDB bandb;
     banmap_t banmap;
+    CNode::SetBannedSetDirty(false);
```

Comment:
> Otherwise there will always be a possible race condition that someone is adding a ban after CNode::SetBannedSetDirty(false); and before bandb.Write(banmap) or CNode::GetBanned(banmap).

That wouldn't be a race condition, but the way it's structured on purpose: because the change to the ban map will re-mark the map as dirty, the next write will catch it. This is not the case right now.

Putting it in a huge lock would hold up the entire P2P for the time of the I/O, which would be unfortunate.